Makefile

```diff
@@ -549,7 +549,7 @@ pylint:
 doc: docs
 
 docs:
-	make -C docs html
+	make -C docs linkcheck doctest html
```

Comment:
The CI process for the website build doesn't use `make docs`. It uses `make html` directly from the docs folder. So this wouldn't impact CI for the website.

If the purpose is for development testing, that's fine, but it would be good to talk more about what the impact of adding these is and what should be done to resolve the outputs of these additional tests. 

We have a link checker that runs separately in CI, and there are discussions on upgrading it. If the intention was to switch this sphinx link checker on in CI, then there needs to be more discussion first.

I'm not clear on the impact of `doctest` and how it is planned to be used, or if it will trigger build blocking errors if added to CI. Running it without `linkcheck` fails. What's it supposed to do?